packages/sources/kaiko-state/src/transport/state.ts

```diff
@@ -0,0 +1,149 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { sleep, makeLogger } from '@chainlink/external-adapter-framework/util'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { BaseEndpointTypes } from '../endpoint/state'
+
+import { generateCreds, getProviderIndicatedTimeUnixMs } from './utils'
+import { ClientReadableStream, ServiceError } from '@grpc/grpc-js'
+import { BehaviorSubject, distinctUntilChanged, from, fromEvent, switchMap } from 'rxjs'
+import { backOff } from 'exponential-backoff'
+
+import { StreamAggregatedStatePriceServiceV1Client } from '@kaiko-data/sdk-node/sdk/sdk_grpc_pb'
+import { StreamAggregatedStatePriceRequestV1 } from '@kaiko-data/sdk-node/sdk/stream/aggregated_state_price_v1/request_pb'
+import { StreamAggregatedStatePriceResponseV1 } from '@kaiko-data/sdk-node/sdk/stream/aggregated_state_price_v1/response_pb'
+
+const logger = makeLogger('KaikoStateTransport')
+
+export type KaikoStateTransportTypes = BaseEndpointTypes
+
+export class KaikoStateTransport extends SubscriptionTransport<KaikoStateTransportTypes> {
+  async initialize(
+    dependencies: TransportDependencies<KaikoStateTransportTypes>,
+    adapterSettings: KaikoStateTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this._start(adapterSettings.API_ENDPOINT, adapterSettings.API_KEY)
+  }
+
+  async _start(endpoint: string, apiKey: string) {
+    const client = new StreamAggregatedStatePriceServiceV1Client(endpoint, generateCreds(apiKey))
+    const request = new StreamAggregatedStatePriceRequestV1()
+    request.setAssetsList(['*'])
```

Comment:
Is it not possible to only request assets in the EA subscription set?

Reply:
It is possible however will make this more complex, as we can't add asset on the fly, we'll need to stop the stream and re-start with new set of assets.

Also there is currently only ~10 assets from the stream.